### ISSUE TITLE
Getting ManagedThreadFactory from JNDI is failing in 20.0.0.9

### ISSUE THREAD

**Issue by jhanders34** [OpenLiberty/open-liberty#13861]
**Describe the bug**
Getting ManagedThreadFactory from JNDI is failing in 20.0.0.9.  This was found by the BPM team.

CWWKN0008E: An object could not be obtained for name concurrent/BPMManagedThreads.
CWWKE0701E: FrameworkEvent ERROR org.osgi.framework.ServiceException: com.ibm.ws.resource.internal.ResourceFactoryTrackerData$1.getService() returned a service object that is not an instance of the service class jakarta.enterprise.concurrent.ManagedThreadFactory
	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:247)
	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:114)
	at org.eclipse.osgi.internal.serviceregistry.ServiceConsumer$2.getService(ServiceConsumer.java:48)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:530)
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getService(ServiceRegistry.java:494)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.getService(BundleContextImpl.java:685)
	at com.ibm.ws.jndi.internal.WSContextFactory.lambda$getService$0(WSContextFactory.java:77)
	at com.ibm.ws.jndi.internal.WSContextFactory$$Lambda$25.000000001B46BA00.run(Unknown Source)
	at java.security.AccessController.doPrivileged(AccessController.java:678)
	at com.ibm.ws.jndi.internal.WSContextFactory.lambda$getService$1(WSContextFactory.java:77)
	at com.ibm.ws.jndi.internal.WSContextFactory$$Lambda$24.000000001B0440E0.apply(Unknown Source)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1671)
	at com.ibm.ws.jndi.internal.WSContextFactory.getService(WSContextFactory.java:77)
	at com.ibm.ws.jndi.internal.WSContext.getReference(WSContext.java:520)
	at com.ibm.ws.jndi.internal.WSContext.resolveObject(WSContext.java:139)
	at com.ibm.ws.jndi.internal.WSContext.lookup(WSContext.java:309)
	at com.ibm.ws.jndi.WSContextBase.lookup(WSContextBase.java:61)
	at org.apache.aries.jndi.DelegateContext.lookup(DelegateContext.java:149)
	at javax.naming.InitialContext.lookup(InitialContext.java:428)
	at com.ibm.test.TestServlet.doGet(TestServlet.java:43)

**Steps to Reproduce**
Code that is failing:

              InitialContext ic = new InitialContext();
              Object threadFactory = ic.lookup("concurrent/BPMManagedThreads");

**Expected behavior**
Should be able to get the ThreadFactory as before.  This is a regression.

**Diagnostic information:**
 - OpenLiberty Version: 20.0.0.9

server.xml snippet to configure this scenario.

```
      <featureManager>
           <feature>webProfile-7.0</feature>
           <feature>concurrent-1.0</feature>
           ...
      </featureManager>
	
	<managedThreadFactory jndiName="concurrent/BPMManagedThreads">
		<contextService>
			<jeeMetadataContext/>
			<classloaderContext/>
			<securityContext/>				
		</contextService>
	</managedThreadFactory>
```

**Comment by jhanders34** [comment]
#regression

**Comment by jhanders34** [comment]
The PR that caused this regression was #13226

**Comment by vaibhavsaxena619** [comment]
Hi,
I am getting the same error. Were you able to debug it?

**Comment by jhanders34** [comment]
> Hi,
> I am getting the same error. Were you able to debug it?

Yes.  The fix was done with PR #13855.
